docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,277 @@
+
+```{note}
+Avoid using jQuery and undocumented jQuery plugins, as they will be removed in a future version of Wagtail.
+```
+
+(extending_client_side_injecting_javascript)=
+
+## Adding custom JavaScript
+
+Within Wagtail's admin interface, there are a few ways to add JavaScript.
+
+The simplest way is to add global JavaScript files via hooks, see [](insert_editor_js) and [](insert_global_admin_js).
+
+For JavaScript added when a specific Widget is used you can add an inner `Media` class to ensure that the file is loaded when the widget is used, see [Django's docs on their form `Media` class](https://docs.djangoproject.com/en/stable/topics/forms/media/#assets-as-a-static-definition).
+
+In a similar way, Wagtail's [template components](template_components) provide a `media` property or `Media` class to add scripts when rendered.
+
+These will ensure the added files are used in the admin after the core JavaScript admin files are already loaded.
+
+(extending_client_side_using_events)=
+
+## Extending with DOM events
+
+When approaching client-side customisations or adopting new components, try to keep the implementation simple first, you may not need any knowledge of Stimulus, React, JavaScript Modules or a build system to achieve your goals.
+
+The simplest way to attach behaviour to the browser is via [DOM Events](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events) and plain (vanilla) JavaScript.
+
+### Wagtail's custom DOM events
+
+Wagtail supports some custom behaviour via listening or dispatching custom DOM events.
+
+-   See [Images title generation on upload](images_title_generation_on_upload).
+-   See [Documents title generation on upload](docs_title_generation_on_upload).
+
+(extending_client_side_stimulus)=
+
+## Extending with Stimulus
+
+Wagtail uses [Stimulus](https://stimulus.hotwired.dev/) as a way to provide lightweight client-side interactivity or custom JavaScript widgets within the admin interface.
+
+The key benefit of using Stimulus is that your code can avoid the need for manual initialisation when widgets appear dynamically, such as within modals, `InlinePanel`, or `StreamField` panels.
+
+The [Stimulus handbook](https://stimulus.hotwired.dev/handbook/introduction) is the best source on how to work with and understand Stimulus.
+
+### Adding a custom Stimulus controller
+
+Wagtail exposes two client-side globals for using Stimulus.
+
+1. `window.stimulusApp` the an instance of the core admin Stimulus application.
+2. `window.Stimulus` Stimulus module as exported from `@hotwired/stimulus`.
+
+You need to first create a custom [Stimulus controller](https://stimulus.hotwired.dev/reference/controllers) which will the base `window.Stimulus.Controller` using class inheritance. If you are using a build tool you can extend `import { Controller } from '@hotwired/stimulus';`.
```

Comment:
"which will the base `window.Stimulus.Controller` using class inheritance." => "which should extend `window.Stimulus.Controller` using class inheritance."

Reply:
Thanks, went with a slightly different wording of this whole paragraph, hopefully reads better now.